### ISSUE TITLE
Rule S3626: "Jump statements should not be redundant" issues false positive with try/catch/finally clause

### ISSUE THREAD

**Issue by ahoisl** [SonarSource/sonar-csharp#1265]
### Description

Rule S3626 produces a false positive issue with try/catch/finally clauses, see below example.

### Repro steps

```csharp
try
{
    Console.WriteLine("Try do something");
}
catch (Exception)
{
    Console.WriteLine("do something on fail before cleanup");
    return; // produces warning for S3626 despite code line at very end of snippet
}
finally
{
    Console.WriteLine("Clean up");
}

Console.WriteLine("do something on success after cleanup");
```

The jump statement is not redundant as there is a statement after the try/catch/finally that would only be executed if no exception is thrown.

### Expected behavior

No issues raised

### Actual behavior

Issue S3626 is raised.

### Known workarounds

None.

### Related information

SonarC# Version = 6.7.1
Visual Studio Version = VS 2017

**Comment by valhristov** [comment]
The representation of the try/catch/finally blocks in our CFG implementation is rather limited and that's the reason for this false positive. Related ticket #621
